hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/MessageFlyweight.java

```diff
@@ -175,7 +175,11 @@ public String getStringUtf8() {
     }
 
     public Data getData() {
-        return new HeapData(getByteArray());
+        final int length = buffer.getInt(index + offset);
+        index += Bits.INT_SIZE_IN_BYTES;
+        Data data = new ByteBufferHeapData(buffer.byteArray(), index + offset, length);
```

Comment:
If yes, then there's another trade-off when `ByteBufferHeapData`  is stored in map/cache.`ByteBufferHeapData` will keep a larger `byte[]` than expected and consume more memory.